pkg/engine/engine.go

```diff
@@ -108,12 +168,47 @@ func filterDetectors(filterFunc func(detectors.Detector) bool, input []detectors
 	return output
 }
 
+func (e *Engine) setFoundResults() {
+	atomic.StoreUint32(&e.foundResults, 1)
+}
+
+// HasFoundResults returns true if any results are found.
+func (e *Engine) HasFoundResults() bool {
+	return atomic.LoadUint32(&e.foundResults) == 1
+}
```

Comment:
Why would we need it to be reset? I might be missing something here 🤔 

My current thoughts on this are:

Firstly, although the race condition in this case may not directly lead to an incorrect value, it can still lead to unpredictable behavior, as the Go memory model does not guarantee that writes in one goroutine are immediately visible to other goroutines without synchronization.

Secondly, while in this specific case the race condition may seem harmless, it's generally a good practice to avoid race conditions altogether because they can lead to hard-to-detect bugs and make your code harder to reason about.

Reply:
> Why would we need it to be reset?

We wouldn't (I think we agree on that).

My thought is that even if the operations aren't atomic (or protected by a mutex), it would still work just fine because it *is* idempotent. It's okay to operate on slightly stale data. Though, we are probably only ever calling `HasFoundResults()` *after* waiting for the engine to `Finish()`.

Just thought experiments though.

I only ask that we change the variable name because seeing `foundResults uint32` makes it look like it's the count of results, not a flag.

Reply:
Another idea - we could change it count the results:

```go
// maybe change name from setFoundResults()
func (e *Engine) setFoundResults() { atomic.AddUint32(&e.foundResults, 1) }
func (e *Engine) HasFoundResults() bool { return atomic.LoadUint32(&e.foundResult) > 0 }
```